src/lifetracker/logic/CommandLineResult.java

```diff
@@ -44,87 +32,62 @@ public void setComment(String newComment) {
-        } else {
-            record.add(deadline.format(DateTimeFormatter.ofLocalizedDateTime(DATE_STYLE, TIME_STYLE)));
-
-            if (period == null) {
-                record.add("");
-            } else {
-                record.add(convert(period));
-            }
-        }
-        
-        taskList.add(record);
+    public List<LogicFloating> getFloatingList() {
+        return floatingList;
     }
 
     @Override
-    public void addEventLine(int id, String name, boolean isActive, LocalDateTime start, LocalDateTime end,
+    public void addTaskLine(int id, String name, LocalDateTime deadline, boolean isOverdue, boolean isDone,
             TemporalAmount period) {
-        List<String> record = new ArrayList<>();
-        record.add(String.valueOf(id));
-        record.add(name);
-        record.add(Boolean.toString(isActive));
-        record.add(start.format(DateTimeFormatter.ofLocalizedDateTime(DATE_STYLE, TIME_STYLE)));
-        record.add(end.format(DateTimeFormatter.ofLocalizedDateTime(DATE_STYLE, TIME_STYLE)));
-
-        if (period == null) {
-            record.add("");
+        if (deadline == null) {
+            addFloatingLine(id, name, isDone);
         } else {
-            record.add(convert(period));
+            addDeadlineLine(id, name, deadline, isOverdue, isDone, period);
         }
-
-        eventList.add(record);
     }
 
-    public String convert(TemporalAmount temporalAmount) {
-        if (temporalAmount != null) {
-            if (temporalAmount.equals(Period.ZERO) || temporalAmount.equals(Duration.ZERO)) {
-                return "";
-            } else if (temporalAmount instanceof Period) {
-                return convertPeriodToString(((Period) temporalAmount).normalized());
-            } else {
-                return convertDurationToString((Duration) temporalAmount);
-            }
-        }
-        return "";
-    }
-
-    private String convertPeriodToString(Period period) {
-        int years = period.getYears();
-        int months = period.getMonths();
-        int days = period.getDays();
-
-        return formatDuration(years, YEAR_FIELD) + formatDuration(months, MONTH_FIELD) + formatDuration(days, DAY_FIELD);
+    private void addDeadlineLine(int id, String name, LocalDateTime deadline, boolean isOverdue, boolean isDone,
+            TemporalAmount period) {
+        LogicDeadline record = new LogicDeadlineImpl();
+        record.setId(id);
+        record.setName(name);
+        record.setDeadline(deadline);
+        record.setOverdue(isOverdue);
+        record.setDone(isDone);
+        record.setPeriod(period);
+        deadlineList.add(record);
     }
 
-    private String convertDurationToString(Duration duration) {
-        long hours = duration.toHours();
-        long minutes = duration.toMinutes() % 60;
-
-        return formatDuration(hours, HOUR_FIELD) + formatDuration(minutes, MINUTE_FIELD);
+    private void addFloatingLine(int id, String name, boolean isDone) {
```

Comment:
O.o, how do you plan to differentiate floating tasks from non-floating tasks in the main Logic class?